schemas/2015-08-01/Microsoft.Cache.json

```diff
@@ -1,173 +1,210 @@
-                }
-              ],
-              "description": "Microsoft.Cache/Redis/redisConfiguration"
+            {
+              "$ref": "https://schema.management.azure.com/schemas/common/definitions.json#/definitions/expression"
+            }
+          ],
+          "description": "All Redis Settings. Few possible keys: rdb-backup-enabled,rdb-storage-connection-string,rdb-backup-frequency,maxmemory-delta,maxmemory-policy,notify-keyspace-events,maxmemory-samples,slowlog-log-slower-than,slowlog-max-len,list-max-ziplist-entries,list-max-ziplist-value,hash-max-ziplist-entries,hash-max-ziplist-value,set-max-intset-entries,zset-max-ziplist-entries,zset-max-ziplist-value etc."
+        },
+        "enableNonSslPort": {
+          "oneOf": [
+            {
+              "type": "boolean"
             },
-            "virtualNetwork": {
-              "oneOf": [
-                {
-                  "$ref": "#/definitions/virtualNetworkId"
-                },
-                {
-                  "$ref": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#/definitions/expression"
-                }
-              ],
-              "description": "Microsoft.Cache/Redis (optional) the ARM resource ID of a Classic Virtual Network for the redis cache to join"
+            {
+              "$ref": "https://schema.management.azure.com/schemas/common/definitions.json#/definitions/expression"
+            }
+          ],
+          "description": "If the value is true, then the non-SLL Redis server port (6379) will be enabled."
+        },
+        "tenantSettings": {
+          "oneOf": [
+            {
+              "type": "object",
+              "additionalProperties": {
+                "type": "string"
+              },
+              "properties": {}
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/common/definitions.json#/definitions/expression"
+            }
+          ],
+          "description": "tenantSettings"
+        },
+        "shardCount": {
+          "oneOf": [
+            {
+              "type": "integer"
+            },
+            {
+              "$ref": "https://schema.management.azure.com/schemas/common/definitions.json#/definitions/expression"
+            }
+          ],
+          "description": "The number of shards to be created on a Premium Cluster Cache."
+        },
+        "virtualNetwork": {
+          "type": "string",
+          "description": "The exact ARM resource ID of the virtual network to deploy the Redis cache in. Example format: /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/Microsoft.ClassicNetwork/VirtualNetworks/vnet1"
+        },
+        "subnet": {
+          "type": "string",
+          "description": "Required when deploying a Redis cache inside an existing Azure Virtual Network."
+        },
+        "staticIP": {
```

Comment:
Shouldn't virtualNetwork, subnet, and staticIP also really be declared as oneOf with 'expression'? So expressions can be used in valid ARM templates? i.e.:
            {
              "$ref": "https://schema.management.azure.com/schemas/common/definitions.json#/definitions/expression"
            }

Reply:
See previous comment - technically an ARM expression will also pass this validation.